packages/react-native/ReactCommon/react/nativemodule/core/platform/android/ReactCommon/JavaTurboModule.cpp

```diff
@@ -107,16 +139,92 @@ jni::local_ref<JCxxCallbackImpl::JavaPart> createJavaCallbackFromJSIFunction(
                     jsi::valueFromDynamic(strongWrapper2->runtime(), val));
               }
 
-              strongWrapper2->callback().call(
-                  strongWrapper2->runtime(),
-                  (const jsi::Value*)args.data(),
-                  args.size());
+              callbackExecutor(strongWrapper2->runtime(), strongWrapper2->callback(), args);
             });
 
         wrapperWasCalled = true;
       });
 }
 
+jni::local_ref<JCxxCallbackImpl::JavaPart> createJavaCallbackFromJSIFunction(
+    jsi::Function &&function,
+    jsi::Runtime &rt,
+    const std::shared_ptr<CallInvoker> &jsInvoker) {
+    CallbackExecutor executor = [](
+            jsi::Runtime& runtime,
+            jsi::Function& callback,
+          const std::vector<jsi::Value>& args) {
+      callback.call(
+              runtime,
+              (const jsi::Value *)args.data(),
+              args.size());
+  };
+  return createJavaCallbackFromJSIFunction(std::move(function), rt, jsInvoker, std::move(executor));
```

Comment:
Avoid this, and just check for an empty `callbackExecutor` the original method, and execute `callback.call` directly.